ui/analyse/src/view/components.ts

```diff
@@ -383,12 +384,61 @@ export const renderMaterialDiffs = (ctrl: AnalyseCtrl): [VNode, VNode] =>
 export const addChapterId = (study: StudyCtrl | undefined, cssClass: string) =>
   cssClass + (study && study.data.chapter ? '.' + study.data.chapter.id : '');
 
+function broadcastChatHandler(ctrl: AnalyseCtrl): BroadcastChatHandler {
+  const encodeMsg = (text: string): string => {
+    if (ctrl.study?.relay && !ctrl.study.relay.tourShow()) {
+      const chapterId = ctrl.study.currentChapter().id;
+      const ply = ctrl.study.currentNode().ply;
+      // '\ue666' was arbitrarily chosen from the unicode private use area to separate the text from the chapterId and ply
+      text = text + '\ue666' + chapterId + '\ue666' + ply;
+    }
+    return text;
+  };
+
+  const cleanMsg = (msg: string): string => {
+    if (msg.includes('\ue666') && ctrl.study?.relay) {
+      return msg.split('\ue666')[0];
+    }
+    return msg;
+  };
+
+  const jumpToMove = (msg: string): void => {
+    if (msg.includes('\ue666') && ctrl.study?.relay) {
+      const segs = msg.split('\ue666');
+      if (segs.length == 3) {
+        const [_, chapterId, ply] = segs;
+        ctrl.study.setChapter(chapterId);
+        setTimeout(() => ctrl.jumpToMain(parseInt(ply)), 100);
```

Comment:
It doesn't take to the correct move for some reason without the setTimeout.

Reply:
Probably because there's network in between. The 100ms hack will not always work depending on who uses it and where from.